### PR TITLE
Add basic ui component structure using the MVP pattern

### PR DESCRIPTION
**Pull request by Marcarrian**
Github liest wohl von rechts nach links beim Erstellen eines PRs. Sollte alles gut sein jetzt

### CODE REVIEW

**Comment by bbortt** [comment]
what's the advantage of using interfaces for views and listeners?

**Comment by Marcarrian** [comment]
Ich habe mich einfach nach den Folien mit dem Beispiel des Rechners gerichtet.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/FooterPresenter.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package ch.bfh.bti7081.s2020.blue.presenter;
+
+import ch.bfh.bti7081.s2020.blue.view.footer.FooterView;
+
+public class FooterPresenter implements FooterView.FooterViewListener {
```
Comment (calesanz) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/FooterPresenter.java:5]: I think for simple Views the Presenter can be omitted.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/MainView.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+@Route
+@PWA(name = "Social Anxiety Application",
+    shortName = "Social Anxiety App",
+    description = "This is an application to help people with social anxiety.",
+    enableInstallPrompt = false)
+public class MainView extends VerticalLayout {
+
+  public MainView() {
+    Object o = new Object();
```
Comment (calesanz) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/MainView.java:27]: Should this be replaced with a global application state object? Or is it just for demonstration purposes?

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/view/footer/FooterView.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package ch.bfh.bti7081.s2020.blue.view.footer;
```
Comment (calesanz) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/view/footer/FooterView.java:1]: I think having a separate Interface and an Impl class for trivial views is too much. These can be directly implemented.

Comment (bbortt) [reply source/src/main/java/ch/bfh/bti7081/s2020/blue/view/footer/FooterView.java:1]: actually for all views I think..

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/AchievementListPresenter.java [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private AchievementListView view;
+
+  public AchievementListPresenter(Object model, AchievementListView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/AchievementListPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/ChallengeDetailPresenter.java [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private ChallengeDetailView view;
+
+  public ChallengeDetailPresenter(Object model, ChallengeDetailView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/ChallengeDetailPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/ChallengeListPresenter.java [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private ChallengeListView view;
+
+  public ChallengeListPresenter(Object model, ChallengeListView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/ChallengeListPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/FooterPresenter.java [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private FooterView view;
+
+  public FooterPresenter(Object model, FooterView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/FooterPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/HeaderPresenter.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private HeaderView view;
+
+  public HeaderPresenter(Object model, HeaderView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/HeaderPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/JournalListPresenter.java [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private JournalListView view;
+
+  public JournalListPresenter(Object model, JournalListView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/JournalListPresenter.java:15]: no spacing.

File: source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/TherapistListPresenter.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+  private Object model;
+  private TherapistListView view;
+
+  public TherapistListPresenter(Object model, TherapistListView view) {
+    this.model = model;
+    this.view = view;
+    view.addListener(this);
+  }
+
```
Comment (bbortt) [inline source/src/main/java/ch/bfh/bti7081/s2020/blue/presenter/TherapistListPresenter.java:15]: no spacing.

**Pull request merged**